ethclient/ethclient.go

```diff
@@ -490,6 +491,35 @@ func (ec *Client) SendTransaction(ctx context.Context, tx *types.Transaction) er
 	return ec.c.CallContext(ctx, nil, "eth_sendRawTransaction", common.ToHex(data))
 }
 
+// PreparePrivateTransaction sends the encoded raw transaction to Constellation,
+// returning the encoded commitment transaction.
+func (ec *Client) PreparePrivateTransaction(ctx context.Context, encodedTx hexutil.Bytes, privateFrom string, privateFor []string) (hexutil.Bytes, error) {
+	if len(privateFor) == 0 {
```

Comment:
The privateFor object is NULL only for public transactions , but for transactions to itself it has a blank value. Checking length for zero will prevent a node sending private transactions to itself. So please change the check to != NULL rather..

Reply:
changed to allow transaction to self node.